tiledb/sm/rest/rest_profile.h

```diff
@@ -104,6 +105,45 @@ class RestProfile {
   /** Destructor. */
   ~RestProfile() = default;
 
+  /**
+   * Returns an optional RestProfile with the given name, which has value _iff_
+   * it has been saved to the local file; `std::nullopt` otherwise.
+   *
+   * @note This API will _not_ parse the `cloud.json` path. This method is
+   * expected to be used _primarily_ by a `Config` object inheriting
+   * written-parameters off of a `RestProfile`. Because a `RestProfile` is
+   * immutable, this method will guarantee its state (validity) at loadtime
+   * (the top of its usage stack).
+   *
+   * @param name The name of the profile to load.
+   * @param homedir The user's $HOME directory, or desired in-test path.
+   * @return The RestProfile with the given name, iff it's been saved.
+   */
+  static inline std::optional<RestProfile> load_if_exists(
+      const std::string& name, const std::string& homedir) {
+    auto filepath = homedir + constants::rest_profile_filepath;
+
+    // If the local file exists, return the profile of the given name, if
+    // exists.
+    if (std::filesystem::exists(filepath)) {
+      json data;
+      {
+        std::ifstream file(filepath);
+        try {
+          file >> data;
+        } catch (...) {
```

Comment:
`catch (...)` without re-throwing can run into some pretty spectacular trouble in systems with more complicated control flows.  Typical exceptions will derive from `std::exception` so catch that instead.

We should also include `std::exception::what()` in the error message here. If we have to help a customer figure out why they're seeing this error message then we want to avoid a cycle of figuring out how to get that detail.